pilot/pkg/security/model/authentication.go

```diff
@@ -104,41 +105,11 @@ func ConstructSdsSecretConfigForGatewayListener(name, sdsUdsPath string) *auth.S
 }
 
 // ConstructSdsSecretConfig constructs SDS Sececret Configuration for workload proxy.
-func ConstructSdsSecretConfig(name, sdsUdsPath string, metadata *model.NodeMetadata) *auth.SdsSecretConfig {
+func ConstructSdsSecretConfig(name, sdsUdsPath string) *auth.SdsSecretConfig {
 	if name == "" || sdsUdsPath == "" {
 		return nil
 	}
 
-	gRPCConfig := &core.GrpcService_GoogleGrpc{
-		TargetUri:  sdsUdsPath,
-		StatPrefix: SDSStatPrefix,
-		ChannelCredentials: &core.GrpcService_GoogleGrpc_ChannelCredentials{
-			CredentialSpecifier: &core.GrpcService_GoogleGrpc_ChannelCredentials_LocalCredentials{
-				LocalCredentials: &core.GrpcService_GoogleGrpc_GoogleLocalCredentials{},
-			},
-		},
-	}
-
-	// If metadata.SdsTokenPath is non-empty, envoy will fetch tokens from metadata.SdsTokenPath.
-	// Otherwise, if useK8sSATrustworthyJwt is set, envoy will fetch and pass k8s sa trustworthy jwt(which is available for k8s 1.12 or higher),
-	// pass it to SDS server to request key/cert.
-	gRPCConfig.CredentialsFactoryName = FileBasedMetadataPlugName
-	if sdsTokenPath := metadata.SdsTokenPath; len(sdsTokenPath) > 0 {
```

Comment:
With this code being removed, how is the grpc CallCredentials (e.g., the JWT token) configured?

Reply:
it is not being configured. We don't need it, pilot agent already has the jwt token. The reason I removed it is because we cannot use a single cluster with Google grpc and Envoy grpc doesn't support it. which is ok since we don't need it (pending more thorough investigation)